### PR TITLE
feat(b-form-tags): add `ignoreInputFocusSelector` prop to make input focus behavior configurable (closes #5425)

### LINKED ISSUE #5425: Clicking on a b-form-tags does not always result with input getting focus
### Describe the bug

Clicking on the `ul` element inside `b-form-tags` does not cause `input` to get focus, clicking on any other element or the `div.b-form-tags` itself works just fine (it focuses the `input`).

### Steps to reproduce the bug

1. Add tags unitl the `input` breaks to new line
2. Click the white space after the last tag
3. Input does not get focus

### Expected behavior

Clicking anywhere inside the `b-form-tags` component should focus the `input`.

### Versions

**Libraries:**

- BootstrapVue: 2.14.0
- Bootstrap: 4.4.1
- Vue: 2.6.11

**Environment:**

- Device: not relevant
- OS: not relevant
- Browser: not relevant
- Version: not relevant

### Demo link

https://jsfiddle.net/n049f1k5/0

### Additional context

There is a workaround for this: https://jsfiddle.net/n049f1k5/1

### PR DESCRIPTION
**Pull request by paweltatarczuk**
### Describe the PR

This PR adda a new `ignoreInputFocusSelector` prop which defaults to `.b-form-tag`.

With this new prop users are able to define a selector for all elements that should not focus the `<b-form-tags>` input on click.

Closes #5425.

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix (fixes a boo-boo in the code) - `fix(...)`, requires a patch version update
- [x] Feature (adds a new feature to BootstrapVue) - `feat(...)`, requires a minor version update
- [x] Enhancement (augments an existing feature) - `feat(...)`, requires a minor version update
- [ ] ARIA accessibility (fixes or improves ARIA accessibility) - `fix(...)`, requires a patch or minor version update
- [ ] Documentation update (improves documentation or typo fixes) - `chore(docs)`, requires a patch version update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe since breaking changes require a minor version update)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples`, `chore(docs): fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type (patch or minor).**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates
- [ ] Includes component package.json meta section updates (prop, slot and event changes/updates)
- [ ] Includes any needed TypeScript declaration file updates
- [ ] New/updated tests are included and passing (required for new features and enhancements)
- [ ] Existing test suites are passing
- [ ] CodeCov for patch has met target (all changes/updates have been tested)
- [ ] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by tmorehouse** [comment]
This will need to be tested on mobile/touch-based iOS Safari (as some versions of safari have issues with focus events when a click/touch event is not registered between the `<body>` and the event with the focus handler).

@jackmu95 could you test with Safari?

**Comment by tmorehouse** [comment]
The other issue is that users can create their own custom interface, and forcing focus to the input may break their custom interface.

**Comment by paweltatarczuk** [comment]
> The other issue is that users can create their own custom interface, and forcing focus to the input may break their custom interface (unless they prevent focus from bubbling)

Maybe instead of changing the type of event add the `onClick` handler explicitly to `ul` and `div` elements or include these elements in the initial `onClick` condition? But then a custom interface might encounter the same issue.

**Comment by jackmu95** [comment]
The current implementation in this PR completely breaks tab handling when clicking on a tag.

I think we should add the same click handling we currently have on the outer wrapper to the `<ul>` and the "Add button" when in invisible state.

**Comment by jackmu95** [comment]
@paweltatarczuk I tweaked the original click handler to also look from other click events inside the `<b-form-tags>` component and only exclude clicks from or from within `<b-tags>`.

Can you please check out the deployment preview and see if the behavior works for you like it should?

The `onClick()` handler could be tweaked a bit further to exclude other elements that are likely to be able to gain focus. Maybe we should make this configurable by a prop.

**Comment by paweltatarczuk** [comment]
> Can you please check out the deployment preview and see if the behavior works for you like it should?'

Works great for my case!

> The onClick() handler could be tweaked a bit further to exclude other elements that are likely to be able to gain focus. Maybe we should make this configurable by a prop.

Maybe don't re-focus when clicked element has already a focus?

```
if (... evt.target !== document.activeElement)
```

**Comment by tmorehouse** [comment]
Or filter out elements such as `button`, `input`, `select`, `a`? or elements with a tabindex set?

**Comment by jackmu95** [comment]
I've added to check to skip the input focus when the event target has focus and the new `ignoreInputFocusSelector` prop matches a element (defaults to `'.b-form-tag'`).

**Comment by tmorehouse** [comment]
Would this be a breaking change for people who might have a custom render content?

File: src/components/form-tags/form-tags.js [hunk 1]
```diff
@@ -178,6 +186,10 @@ export const BFormTags = /*#__PURE__*/ Vue.extend({
       type: Boolean,
       default: false
     },
+    ignoreInputFocusSelector: {
+      type: [Array, String],
+      default: '.b-form-tag'
+    },
```
Comment (tmorehouse) [inline src/components/form-tags/form-tags.js:192]: should this maybe include elements like `button` ,`input`, and ` select`

**Comment by jackmu95** [comment]
Thanks @paweltatarczuk!

**Pull request merged**